mce-hal.c

```diff
@@ -55,27 +55,29 @@ product_id_t get_product_id(void)
 			product_id = PRODUCT_UNKNOWN;
 		}
 		g_free(tmp);
-		goto EXIT;
+		return product_id;
 	}
 
+	/*
+	 * NOTE: "Nokia N810" is substring of "Nokia N810 WiMAX",
+	 * so must be checked after
+	 */
 	if (mce_read_string_from_file(CPUINFO_PATH, &tmp)) {
 		if (strstr(tmp, "Nokia 770") != NULL) {
 			product_id = PRODUCT_SU18;
 		} else if (strstr(tmp, "Nokia N800") != NULL) {
 			product_id = PRODUCT_RX34;
 		} else if (strstr(tmp, "Nokia N810 WiMAX") != NULL) {
 			product_id = PRODUCT_RX48;
-		} else if (strstr(tmp, "Nokia N810") != NULL) { /* NOTE: "Nokia N810" is substring of "Nokia N810 WiMAX" so must be checked after */
```

Comment:
Why is this removed?

Reply:
Or rather, moved.

Reply:
This comment made the line to be 130+ chars long.
After I moved it to another (appropriate) place, this line doesn't exceed 80 chars now.
Which I think is good. Don't you?

Reply:
I think the comment would be better in the context, so maybe above the `} else if (strstr(tmp, "Nokia N810") != NULL)` line.

Reply:
For me it will become uggly. Compare.
1) Now:
```
        /*
         * NOTE: "Nokia N810" is substring of "Nokia N810 WiMAX",
         * so must be checked after
         */
        if (mce_read_string_from_file(CPUINFO_PATH, &tmp)) {
                if (strstr(tmp, "Nokia 770") != NULL) {
                        product_id = PRODUCT_SU18;
                } else if (strstr(tmp, "Nokia N800") != NULL) {
                        product_id = PRODUCT_RX34;
                } else if (strstr(tmp, "Nokia N810 WiMAX") != NULL) {
                        product_id = PRODUCT_RX48;
                } else if (strstr(tmp, "Nokia N810") != NULL) {
                        product_id = PRODUCT_RX44;
                } else if (strstr(tmp, "Nokia RX-51 board") != NULL) {
                        product_id = PRODUCT_RX51;
                } else {
                        product_id = PRODUCT_UNKNOWN;
                }
                g_free(tmp);
        }

```
2) After the change (auto indentation):
```
        if (mce_read_string_from_file(CPUINFO_PATH, &tmp)) {
                if (strstr(tmp, "Nokia 770") != NULL) {
                        product_id = PRODUCT_SU18;
                } else if (strstr(tmp, "Nokia N800") != NULL) {
                        product_id = PRODUCT_RX34;
                } else if (strstr(tmp, "Nokia N810 WiMAX") != NULL) {
                        product_id = PRODUCT_RX48;
                        /*
                         * NOTE: "Nokia N810" is substring of "Nokia N810 WiMAX",
                         * so must be checked after
                         */
                } else if (strstr(tmp, "Nokia N810") != NULL) {
                        product_id = PRODUCT_RX44;
                } else if (strstr(tmp, "Nokia RX-51 board") != NULL) {
                        product_id = PRODUCT_RX51;
                } else {
                        product_id = PRODUCT_UNKNOWN;
                }
                g_free(tmp);
        }
```

3) After the change (manual indentation):
```
        if (mce_read_string_from_file(CPUINFO_PATH, &tmp)) {
                if (strstr(tmp, "Nokia 770") != NULL) {
                        product_id = PRODUCT_SU18;
                } else if (strstr(tmp, "Nokia N800") != NULL) {
                        product_id = PRODUCT_RX34;
                } else if (strstr(tmp, "Nokia N810 WiMAX") != NULL) {
                        product_id = PRODUCT_RX48;
                /*      
                 * NOTE: "Nokia N810" is substring of "Nokia N810 WiMAX",
                 * so must be checked after
                 */
                } else if (strstr(tmp, "Nokia N810") != NULL) {
                        product_id = PRODUCT_RX44;
                } else if (strstr(tmp, "Nokia RX-51 board") != NULL) {
                        product_id = PRODUCT_RX51;
                } else {
                        product_id = PRODUCT_UNKNOWN;
                }
                g_free(tmp);
        }
```

Reply:
Do you still want this change?

Reply:
I think the manual indentation looks good. If this if statement ever grows to >50+ devices, it's also better to have the comment closer to the line it relates to.

Reply:
Such variant will have auto indentation in place and the comment is placed in appropriate context:
```
        if (mce_read_string_from_file(CPUINFO_PATH, &tmp)) {
                if (strstr(tmp, "Nokia 770") != NULL) {
                        product_id = PRODUCT_SU18;
                } else if (strstr(tmp, "Nokia N800") != NULL) {
                        product_id = PRODUCT_RX34;
                } else if (strstr(tmp, "Nokia N810 WiMAX") != NULL) {
                        product_id = PRODUCT_RX48;
                } else if (strstr(tmp, "Nokia N810") != NULL) {
                        /*
                         * NOTE: "Nokia N810" is substring of "Nokia N810
                         * WiMAX", so must be checked after
                         */
                        product_id = PRODUCT_RX44;
                } else if (strstr(tmp, "Nokia RX-51 board") != NULL) {
                        product_id = PRODUCT_RX51;
                } else {
                        product_id = PRODUCT_UNKNOWN;
                }
                g_free(tmp);
        }

```
What do you think about it?

Reply:
Or maybe just remove this comment?
I doubt someone will change the order of these if's in the future.

Reply:
I am fine with the manual indentation or the above auto indentation, let's not remove it.

Reply:
Fixed. Please check if you are satisfied with how it's done now.

Reply:
Comment looks good.